app/auto_responder.go

```diff
@@ -4,9 +4,33 @@
 package app
 
 import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+func (a *App) checkIfRespondedToday(createdAt int64, channelId, userId string) bool {
+	// get last post in a calender day sent by user and if it's auto responder post then don't send again
+	y, m, d := time.Unix(createdAt, 0).Date()
+	since := model.GetMillisForTime(time.Date(y, m, d, 0, 0, 0, 0, time.UTC))
+	postList, err := a.Srv().Store.Post().GetPostsByUserInChannelSince(
+		model.GetPostsSinceOptions{ChannelId: channelId, Time: since},
+		userId,
+	)
+	if err != nil {
+		mlog.Error("auto_responder.check_for_auto_respond_today_error", mlog.String("error", err.Error()))
```

Comment:
Return the error above. And return `false, err` from there.

Reply:
I think error is of no use to caller function so logging error here and caller function will send the auto response as usual in case of some error happened at store layer.

Reply:
If the store is failing, it's generally an indication that Mattermost is not able to work properly. In that case, it might be better to not suppress the error, and return it upwards so it gets logged at the call site.